addons/im_livechat/static/src/models/chatbot.js

```diff
@@ -27,6 +27,87 @@ registerModel({
             this.messaging.publicLivechatGlobal.chatWindow.legacyChatWindow.$('.o_livechat_chatbot_end').show();
             this.messaging.publicLivechatGlobal.chatWindow.legacyChatWindow.$('.o_livechat_chatbot_restart').one('click', this.messaging.publicLivechatGlobal.livechatButtonView.onChatbotRestartScript);
         },
+        /**
+         * Processes the step, depending on the current state of the script and the author of the last
+         * message that was typed into the conversation.
+         *
+         * This is a rather complicated process since we have many potential states to handle.
+         * Here are the detailed possible outcomes:
+         *
+         * - Check if the script is finished, and if so end it.
+         *
+         * - If a human operator has taken over the conversation
+         *   -> enable the input and let the operator handle the visitor.
+         *
+         * - If the received step is of type expecting an input from the user
+         *   - the last message if from the user (he has already answered)
+         *     -> trigger the next step
+         *   - otherwise
+         *     -> enable the input and let the user type
+         *
+         * - Otherwise
+         *   - if the the step is of type 'question_selection' and we are still waiting for the user to
+         *     select one of the options
+         *     -> don't do anything, wait for the user to click one of the options
+         *   - otherwise
+         *     -> trigger the next step
+         */
+        processStep() {
+            if (this.shouldEndScript) {
+                this.endScript();
+            } else if (
+                this.currentStep.data.chatbot_step_type === 'forward_operator' &&
+                this.currentStep.data.chatbot_operator_found
+            ) {
+                this._chatbotEnableInput();
```

Comment:
in this commit, this should still be `this.messaging.publicLivechatGlobal.livechatButtonView.widget._chatbotEnableInput();`

Reply:
change done